docs/details/widgets/scale.rst

```diff
@@ -4,110 +4,258 @@
-and :cpp:expr:`lv_obj_set_style_length(scale, 5, LV_PART_ITEMS);` for minor ticks. The ticks can be
-padded in either direction (outwards or inwards) for RADIAL scales only with:
-:cpp:expr:`lv_obj_set_style_radial_offset(scale, 5, LV_PART_INDICATOR);` for major ticks and
-:cpp:expr:`lv_obj_set_style_radial_offset(scale, 5, LV_PART_ITEMS);` for minor .
+    number of ticks and the Scale range, so the label drawn can present rounding
+    errors when the calculated value is a floating-point value.
+
+The length of the ticks can be configured with the length Style property on the
+:cpp:enumerator:`LV_PART_INDICATOR` for major ticks and :cpp:enumerator:`LV_PART_ITEMS`
+for minor ticks.  Example with local Style:
+:cpp:expr:`lv_obj_set_style_length(scale, 5, LV_PART_INDICATOR)` for major ticks
+and :cpp:expr:`lv_obj_set_style_length(scale, 5, LV_PART_ITEMS)` for minor ticks. The ticks can be
+padded in either direction (outwards or inwards) for ``..._ROUND_...`` Scales only with:
+:cpp:expr:`lv_obj_set_style_radial_offset(scale, 5, LV_PART_INDICATOR)` for major ticks and
+:cpp:expr:`lv_obj_set_style_radial_offset(scale, 5, LV_PART_ITEMS)` for minor.
 Using length and radial offset together allows total control of the tick position.
-i.e. :cpp:expr:`static char * custom_labels[3] = {"One", "Two", NULL}`.
-It is also possible to offset the labels from the major ticks (either positive or negative) using 
-:cpp:expr:`lv_obj_set_style_pad_radial(scale, 5, LV_PART_INDICATOR);`
+
+It is also possible to offset the labels from the major ticks (either positive or negative) using
+:cpp:expr:`lv_obj_set_style_pad_radial(scale, 5, LV_PART_INDICATOR)`
+
+
+.. _scale_sections:
 
 Sections
 --------
 
-A section is the space between a minor and a major range. They can be created with :cpp:expr:`lv_scale_add_section(scale)`
-and it handles back an :cpp:type:`lv_scale_section_t` pointer.
+Sections make it possible for portions of a Scale to *convey meaning* by using
+different Style properties to draw them (colors, line thicknesses, font, etc.).
+
+A Section represents a sub-range of the Scale, whose Styles (like Cascading Style
+Sheets) take precedence while drawing the PARTS (lines, arcs, ticks and labels) of
+the Scale that are within the range of that Section.
+
+If a PART of a Scale is within the range of 2 or more Sections (i.e. those Sections
+overlap), the Style's properties belonging to the most recently added Section takes
+precedence over the same style properties of other Section(s) that "involve" that
+PART.
+
+
+.. _scale_creating_sections:
+
+Creating Sections
+~~~~~~~~~~~~~~~~~
+
+A Section is created using :cpp:expr:`lv_scale_add_section(scale)`, which returns a
+pointer to a :cpp:type:`lv_scale_section_t` object.  This creates a Section with
+range [0..0] and no Styles added to it, which ensures that Section will not be drawn
+yet:  it needs both a range inside the Scale's range and at least one :ref:`Style
```

Comment:
That would be good default behavior but it seems like leaving the range uninitialized has an erroneous effect.

Commenting out these lines: https://github.com/lvgl/lvgl/blob/bfc47f424a6df497fdde2b6b706821189038a504/examples/widgets/scale/lv_example_scale_2.c#L77-L80

![image](https://github.com/user-attachments/assets/421361e3-d859-4d35-a89f-ab64b573151d)

Reply:
Wow!  A bug!

### Here is what I found:

From ONE perspective (bear with me), I could say it is the initialization value with range [0..0] that causes that.  If it is initialized to [0..-1] (as discussed in issue #7285 as a "neutral" range, guaranteed not to be drawn yet until user changes it), it comes out like this:

![image](https://github.com/user-attachments/assets/c490e528-d1df-4b46-8d40-0f57be7b53a6)

However....  [0..0] is actually a valid range! (It includes the value 0 only, and *might* be useful in certain contexts.) So I went hunting.  I found some pretty interesting bugs!  All the below are in `lv_scale.c`.

1. I discovered that the "stray" line was caused by a combination of bugs:
    - It started with having 2 uninitialized fields (random values after a `malloc`) in the new `section` objects when they were being added: `last_tick_in_section` and `first_tick_in_section`.  Instead of initializing these 2 fields, I simply cleaned up the `lv_scale_add_section()` function with an `lv_memzero()` (IMO should have been there in the first place), and removed each field assignment to NULL and 0 values, and kept the non-zero values.  Before I did that, the "Y" coordinate of one end of the section had a HUGE negative number from the random data after the `malloc` -- which also caused the vertical stray line on its own.  After it was being set to 0, I found 2 more things that were involved.
    - `scale_find_section_tick_idx()` had a logic error which *assumed that the beginning and ending range values were different*.  When they were the same, `last_tick_idx_in_section` and `last_tick_idx_is_major` did not get set *at all*.  It could have been any range which included a TICK mark, like [10..10] in this example (`lv_example_scale_2.c`).
    - `scale_store_section_line_tick_width_compensation()` had a logic error in it that did not allow for the beginning and ending values of the ranges to be the same.  With this bug fixed, it ensured that `last_tick_idx_in_section` was initialized correctly.  It *not* being initialized correctly had later side effects resulting in the "Y" coordinate of the section being left at 0, which *also* "extended" the section line to near the top of the "screen" in the drawing.

2. There were several places where `unsigned` math was being applied to `signed` integers.  Example:  Line 841, x- and y-offsets are signed integers and yet math is being done with them using `unsigned` integers.  Specifically:
```c
            main_line_point_a.x = x_ofs - 1U;
```
The danger lurking there is contained in `C automatic promotions`.  Specifically, before that subtraction is performed (as well as other types of math operations), `x_ofs` is silently "promoted" to an `unsigned` integer!  So if `x_ofs` is ever 0, this will result in `x` containing the value 0xFFFFFFFF!  Thankfully, `x` is signed, so it "back converts" back to `-1` again.  But this is a coding error.  There are quite a few of these, which I fixed.  They are marked where a constant value (e.g. `0` or `2`) had a 'U' after it causing it to act as an `unsigned` integer in the places where they were being used.  I found some places where if the Scale's coordinates had been slightly negative (partially left of the screen), *all hell would have broken loose* with the negative values being promoted to `unsigned` values before math was performed.

Unfortunately, there were a lot of these, and since I was already aware of them, I couldn't let them stand, so you will see them in the DIFFs as a missing 'U's where they were interfering with intentionally-signed math operations, including as arguments being passed to `lv_map()`.

------

Now, with all those lines commented out, it is correct:

![image](https://github.com/user-attachments/assets/d90e2a7f-0a16-4b63-a51c-d827e11f7f2e)

And the range [0..0] is indeed honored as a valid range.  As you can see with JUST this line running in the example (with range still [0..0]),
```c
    lv_scale_section_set_style(section, LV_PART_INDICATOR, &section_label_style);
```
it looks like this, which is how it is supposed to look.  Note that this was not possible before -- it would have extended the line through the top of the screen.

![image](https://github.com/user-attachments/assets/1d541d3b-e08d-447a-a527-143a33908675)

And showing that I got all the `signed` vs `unsigned` stuff right, this also would not have been possible before, but now it is being correctly illustrated partially off screen to the left.

![image](https://github.com/user-attachments/assets/9f8fdc36-d25e-47e7-8599-6a616ab443fc)

And since this all came up in this PR and they are all now fixed, I am submitting these bug-fixes with this.